src/solvers/smt2_incremental/smt_bitvector_theory.cpp

```diff
@@ -0,0 +1,42 @@
+// Author: Diffblue Ltd.
+
+#include <solvers/smt2_incremental/smt_bitvector_theory.h>
+
+#include <util/invariant.h>
+
+static void validate_bitvector_predicate_arguments(
+  const smt_termt &left,
+  const smt_termt &right)
+{
+  const auto left_sort = left.get_sort().cast<smt_bit_vector_sortt>();
+  INVARIANT(left_sort, "Left operand must have bitvector sort.");
+  const auto right_sort = right.get_sort().cast<smt_bit_vector_sortt>();
+  INVARIANT(right_sort, "Right operand must have bitvector sort.");
+  INVARIANT(
+    left_sort->bit_width() == right_sort->bit_width(),
+    "Left and right operands must have the same bit width.");
+}
```

Comment:
Do we need to check they are both signed/unsigned?

Reply:
Bit vectors in SMT2 bitvector theory do not have signedness as a property. Therefore this information is not available to this factory enforcing these invariants. Good ways to check use of the correct kind of operator might be worth us discussing over slack.